shared/metadata.go

```diff
@@ -111,30 +135,20 @@ func constructMetadataResponse(productSpecs ProductSpecs, metadata map[string]Me
 			link := data.Links[i]
 			for _, result := range link.Results {
 				//TODO(kyleju): Concatenate test path on WPT Metadata repository instead of here.
-				var fullTestName = GetWPTTestPath(folderPath, result.TestPath)
+				fullTestName := GetWPTTestPath(folderPath, result.TestPath)
+
+				// If ProductSpecs is empty, that implies we want test-level issues.
+				//
+				// TODO(smcgruer): Support asking for both test-level issues and product issues at once.
+				if len(productSpecs) == 0 && link.Product.BrowserName == "" {
+					addResponseLink(fullTestName, link, result, res)
+				}
+
+				// Otherwise, find any matching product for this link result (there can be at most one).
 				for _, productSpec := range productSpecs {
-					// Matches browser type if a version is not specified.
-					if link.Product.MatchesProductSpec(productSpec) ||
-						// Matches to all browsers if product is not specified.
-						link.Product.BrowserName == "" {
-						output := MetadataLink{
-							Product: link.Product,
-							URL:     link.URL,
-						}
-						if result.SubtestName != nil || result.Status != nil {
-							output.Results = []MetadataTestResult{
-								{
-									SubtestName: result.SubtestName,
-									Status:      result.Status,
-									// TestPath is redundant (it's the map key)
-								},
-							}
-						}
-						if _, ok := res[fullTestName]; !ok {
-							res[fullTestName] = MetadataLinks{output}
-						} else {
-							res[fullTestName] = append(res[fullTestName], output)
-						}
+					// Matches on browser type if a version is not specified.
+					if link.Product.MatchesProductSpec(productSpec) {
```

Comment:
Without `|| link.Product.BrowserName == "" `, aren't we skipping all the metadata for `test-issue` in `/api/metadata`? But looks like it is still working on [staging](https://smcgruer-test-issue-se-dot-wptdashboard-staging.uk.r.appspot.com/results/css/css-align?label=master&label=experimental&aligned)?

Reply:
Ok, I fixed this by adding an `includeTestLevel` flag, but I could also see adding `test-issue` as an acceptable `product/products` option for `/api/metadata` only. Lmk what you think.